Update CrUX guides with Core Web Vitals (#3061)
GoogleChrome/web.dev: src/site/content/en/fast/chrome-ux-report-data-studio-dashboard/index.md

title: Using the CrUX Dashboard on Data Studio
authors:
  - rviscomi
hero: hero.png
description: |
  Data Studio is a powerful data visualization tool that enables you to build

>   dashboards on top of big data sources, like the Chrome UX Report In this

Reviewer: Drive-by: Missing a period between `Report` and `In`

and the visualizations of Data Studio. It eliminates the need for users of the 
dashboard to write any queries or generate any charts. Everything is built for 
you; all you need is to provide an origin and a custom dashboard will be 
generated for you.

## Creating a dashboard

To get started, go to [g.co/chromeuxdash](https://g.co/chromeuxdash).
This will take you to the CrUX community connector page where you can provide
the origin for which the dashboard will be generated. Note that first-time 
users may need to complete permission or marketing preference prompts.

<img class="w-screenshot" src="./dash_connector.png" alt="CrUX Dashboard connector">

The text input field only accepts origins, not full URLs. For example:

{% Compare 'better', 'Origin (Supported)' %}
```text
https://developers.google.com
```
{% endCompare %}

{% Compare 'worse', 'URL (Not supported)' %}
```text
https://developers.google.com/web/tools/chrome-user-experience-report/
```
{% endCompare %}

If you omit the protocol, HTTPS is assumed. Subdomains matter, for example
`https://developers.google.com` and `https://www.google.com` are considered to 

> be very different origins.

Reviewer: ```suggestion
be different origins.
```
I don't think we want the emphasis of `very` here because it implies that there are shades/degrees of differentness in origins

Some common issues with origins are providing the wrong protocol, for example
`http://` instead of `https://`, and omitting the subdomain when needed.
Some websites include redirects, so if `http://example.com` redirects to
`https://www.example.com`, then you should use the latter, which is the 
canonical version of the origin. As a rule of thumb, use whichever origin 
users see in the URL bar.

If your origin is not included in the CrUX dataset, you may get an error message
like the one below. There are over 4 million origins in the dataset, but the one
you want may not have sufficient data to be included.

<img class="w-screenshot" src="./dash_error.png" alt="CrUX Dashboard error message">

If the origin exists, you'll be taken to the schema page for the dashboard.
This shows you all of the fields that are included: each effective connection
type, each form factor, the month of the dataset release, the distribution of
performance for each metric, and of course the name of the origin. There's

> nothing you need to do or change on this page, just click "Create Report" to 

Reviewer: ```suggestion
nothing you need to do or change on this page, just click **Create Report** to 
```
https://developers.google.com/style/ui-elements#formatting

continue.

<img class="w-screenshot" src="./dash_schema.png" alt="CrUX Dashboard schema">

## Using the dashboard

Each dashboard comes with three types of pages:

1. Core Web Vitals overview
2. Metric performance
3. User demographics

Each page includes a chart showing distributions over time for each available
monthly release. As new datasets are released, you can simply refresh the
dashboard to get the latest data.

The monthly datasets are released on the second Tuesday of every month. For 
example, the dataset consisting of user experience data from the month of May 
is released on the second Tuesday of June.

### Core Web Vitals overview

The first page is an overview of the origin's monthly
[Core Web Vitals](/vitals/) performance. These are the most important UX metrics that Google recommends you focus on.

<img class="w-screenshot" src="./dash_cwv.png" alt="CrUX Dashboard Core Web Vitals overview">

Use the Core Web Vitals page to understand how the origin is experienced by 
desktop and phone users. By default, the most recent month at the time you 
created the dashboard is selected. To change between older or newer monthly 

> releases, use the "Month" filter at the top of the page.

Reviewer: ```suggestion
releases, use the **Month** filter at the top of the page.
```

Note that tablet is omitted from these charts by default, but if needed you 

> could remove the "No Tablet" filter in the bar chart configuration, shown 

Reviewer: ```suggestion
could remove the **No Tablet** filter in the bar chart configuration, shown 
```

below.

<img class="w-screenshot" src="./dash_cwv_tablet.png" alt="Editing the CrUX Dashboard to show tablets on the Core Web Vitals page">

### Metric performance

After the Core Web Vitals page, you'll find standalone pages for all
[metrics](https://developers.google.com/web/tools/chrome-user-experience-report/#metrics)
in the CrUX dataset.

<img class="w-screenshot" src="./dash_lcp.png" alt="CrUX Dashboard LCP page">

> Atop each page is the "Device" filter, which you can use to restrict the form 

Reviewer: ```suggestion
Atop each page is the **Device** filter, which you can use to restrict the form 
```

factors included in the experience data. For example, you can drill down 
specifically into phone experiences. This setting persists across pages.

The primary visualizations on these pages are the monthly distributions of
experiences categorized as "Good", "Needs Improvement", and "Poor". The 
color-coded legend below the chart indicates the range of experiences included
in the category. For example, in the screenshot above, you can see the 
percent of good [Largest Contentful Paint](/lcp/) (LCP) experiences, which are 

> categorized as those less than 2500ms, fluctuating and getting slightly worse 

Reviewer: I recommend omitting a reference to a specific value (`2500ms`) unless you have some sort of system that tracks every page where specific CWV values are hardcoded (so that you can quickly update all those pages if the values ever change in the future)

in recent months.

The most recent month's percentages of good and poor experiences are shown 
above the chart along with an indicator of the percent difference from the 
previous month. For this origin, good LCP experiences fell by 3.2% to 56.04% 
month-over-month.

{% Aside 'caution' %}

> Due to a quirk with Data Studio, you may sometimes see "No Data" here. This is

Reviewer: ```suggestion
Due to a quirk with Data Studio, you may sometimes see `No Data` here. This is
```
I'm not sure if this one is captured in the Style Guide but my convention is to use code formatting in these cases because it's literal text that you see on the screen

normal and due to the previous month's release not being available until the
second Tuesday.
{% endAside %}

Additionally, for metrics like LCP and other Core Web Vitals that provide 
explicit percentile recommendations, you'll find the "P75" metric between the 
good and poor percentages. This value corresponds to the origin's 75th 
percentile of user experiences. In other words, 75% of experiences are better
than this value. One thing to note is that this applies to the overall 
distribution across _all devices_ on the origin. Toggling specific devices 

> with the "Device" filter will not recalculate the percentile.

Reviewer: ```suggestion
with the **Device** filter will not recalculate the percentile.
```

{% Details %}
{% DetailsSummary %}
Boring technical caveats about percentiles
{% endDetailsSummary%}

Be aware that the percentile metrics are based on the histogram data from 
[BigQuery](/chrome-ux-report-bigquery/), so the granularity will be coarse: 
1000ms for LCP, 100ms for FID, and 0.05 for CLS. In other words, a P75 LCP of
3800ms indicates that the true 75th percentile is somewhere between 3800ms and
3900ms.

Additionally, the BigQuery dataset uses a technique called "bin spreading" in 
which densities of user experiences are intrinsically grouped into very coarse 
bins of decreasing granularity. This allows us to include minute densities in 
the tail of the distribution without having to exceed four digits of 
precision. For example, LCP values less than 3 seconds are grouped into bins 

> 200ms wide. Between 3 and 10 seconds, bins are 500ms wide, then 5000ms wide, 
> etc. Rather than having bins of varying widths, bin spreading ensures that all 

Reviewer: ```suggestion
200ms wide. Between 3 and 10 seconds, bins are 500ms wide. Beyond 10 seconds, bins are 5000ms wide, 
etc. Rather than having bins of varying widths, bin spreading ensures that all 
```
I stumbled over this paragraph. If my understanding is correct then I think this phrasing will make it easier to grok on first try

bins are a constant 100ms wide, and the distribution is linearly interpolated 

> across each bin.

Reviewer: We just said that bins go from 200ms wide then 500ms wide then 5000ms wide but here we're saying they are a constant 100ms wide

Since these are technical details you can leave this as is if you think it's a good description of what's going on, I'm just letting you I didn't get it on my first readthrough

Author: Added a parenthetical about GCD

Corresponding P75 values in tools like PageSpeed Insights are not based on the 
public BigQuery dataset and are able to provide millisecond-precision values.
{% endDetails %}

### User demographics

There are two [dimensions](https://developers.google.com/web/tools/chrome-user-experience-report/#dimensions)
included on the user demographic pages: devices and effective connection
types (ECTs). These pages illustrate the distribution of page views across the 
entire origin for users in each demographic.

The device distribution page shows you the breakdown of phone, desktop, and
tablet users over time. Many origins tend to have little to no tablet data so
you'll often see "0%" hanging off the edge of the chart.

<img class="w-screenshot" src="./dash_device.png" alt="CrUX Dashboard device page">

Similarly, the ECT distribution page shows you the breakdown of 4G, 3G, 2G, 
slow 2G, and offline experiences.

{% Aside 'key-term' %}
Effective connection types are considered _effective_ because they're based 
on bandwidth measurements on users' devices, and don't imply any particular 

> technology used. For example, a desktop user on fast WiFi may be labelled as 

Reviewer: ```suggestion
technology used. For example, a desktop user on fast Wi-Fi may be labelled as 
```
